### ISSUE TITLE
Feature request: Implement Promise interface.

### ISSUE THREAD

**Issue by svenvandescheur** [Stanko/animated-scroll-to#23]
It would be very useful to have `animateScrollTo()` return a `Promise` object. This would help a lot in environments where scrolling is part of a bigger animation path.

```
animateScrollTo(node, { offset: -100 })
   .then(node => console.log('scrolled to', node))
```

**Comment by Stanko** [comment]
Hello @svenvandescheur,
That is interesting suggestion. Currently there is `onComplete` param which allows you a callback which will be called when scrolling is complete. I want to rewrite this library, but always something else pops up. I'll have this in mind though!

Cheers!

**Comment by svenvandescheur** [comment]
@Stanko I won't be able to look into it anytime soon but I might in a month or so. I think the biggest issue here is dealing with legacy environments that don't support Promises. My guess is that the actual implementation shouldn't be that difficult.

**Comment by Stanko** [comment]
Released in 2.0.0-beta.0
